erpnext/accounts/doctype/payment_request/payment_request.py

```diff
@@ -33,6 +34,25 @@ def validate_currency(self):
 		if self.payment_account and ref_doc.currency != frappe.db.get_value("Account", self.payment_account, "account_currency"):
 			frappe.throw(_("Transaction currency must be same as Payment Gateway currency"))
 
+	def on_update(self):
+		self.validate_subscription_details()
+
+	def validate_subscription_details(self):
```

Comment:
Hi @saurabh6790,
Thanks a lot for your review. I will make the necessary corrections.
I'm only calling the validation `on_update` because with `validate` the popup is shown a few times in case of amount discrepancy.
Since a discrepancy is not blocking, I think we should only make this verification once to improve user experience.